src/test/kotlin/org/rust/cargo/util/CargoWithToolchainCommandCompletionProviderTest.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.util
+
+import org.rust.cargo.RsWithToolchainTestBase
+import org.rust.cargo.project.model.cargoProjects
+
+class CargoWithToolchainCommandCompletionProviderTest : RsWithToolchainTestBase() {
+    fun `test target triple completion includes popular targets`() {
+        val testProject = buildProject {
+            toml("Cargo.toml", """
+                [package]
+                name = "hello"
+                version = "0.1.0"
+                authors = []
+            """)
+
+            dir("src") {
+                rust("main.rs", """
+                    fn main() {}
+                """)
+            }
+        }
+
+        val cargoProjects = testProject.psiFile("src/main.rs").project.cargoProjects
+        val cargoProject = cargoProjects.allProjects.first()
+        val completions = CargoCommandCompletionProviderTest.complete("build --target ", cargoProjects, cargoProject.workspace)
+
+        assertContainsElements(
+            completions,
+            // Tier 1
+            "aarch64-unknown-linux-gnu",
+            "i686-pc-windows-gnu",
+            "i686-pc-windows-msvc",
+            "i686-pc-windows-msvc",
+            "x86_64-apple-darwin",
+            "x86_64-pc-windows-gnu",
+            "x86_64-pc-windows-msvc",
+            "x86_64-pc-windows-gnu",
+            // Other popular
+            "aarch64-apple-darwin",
```

Comment:
1.41.1 doesn't have this target. Let's exclude it from the list and then all tests will pass